File: TECH/React/router-app-ex1/src/components/Book.js
import React from "react";
import { useParams, Link } from "react-router-dom";

const Book = ({ book }) => {
  const { keyword } = useParams();
  const selectedBook = book.find((b) => b.id === parseInt(keyword));

  return (
    <div>
      <h2>도서 상세 정보</h2>
      {selectedBook ? (
        <div>
          <h3>{selectedBook.title}</h3>
          <p>저자: {selectedBook.author}</p>
          <p>가격: {selectedBook.price}</p>
          <p>설명: {selectedBook.description}</p>
          <Link to="/content">목록으로 돌아가기</Link>
        </div>
      ) : (
        <p>해당 ID의 도서를 찾을 수 없습니다.</p>
      )}
    </div>
  );
};

export default Book;

File: TECH/Java/oopClass/sec09/Book.java
package oopClass.sec09;

public class Book {

  String title;
  String author;
  int price;

  public Book() {
    //this() : 생성자에서 다른 생성자 호출 할 수 있는 메서드
    this("자바스크립트", "이몽룡", 12000); //매개변수로 생성자를 구분
  }

  public Book(String t, String au, int p) {
    title = t;
    author = au;
    price = p;
  }

  public void show() {
    System.out.println("Title: " + title);
    System.out.println("Author: " + author);
    System.out.println("Price: " + price);
  }
}
